06.bowling_object/game.rb

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def calc_score(shot_scores)
+    frames = []
+    shot_scores.gsub('X', '10').split(',').each do |shot_score|
+      apply_bonus(frames, shot_score.to_i)
+      new_frame = create_frame(frames)
+      frames << new_frame if new_frame
+      target_frame = frames.last
+      target_frame.record_shot(shot_score.to_i)
```

Comment:
```suggestion
      apply_bonus(frames, shot_score.to_i)
      target_frame = frames.last
      target_frame.record_shot(shot_score.to_i)
```

apply_bonusはこの位置でも動くはずです

仮に`apply_bonus`メソッドを使わないとするとこうなります。

```suggestion
      frames.each do |frame|
        frame.add_bonus(shot_score.to_i)
      end
      target_frame = frames.last
      target_frame.record_shot(shot_score.to_i)
```

さらに、lastかどうかにかかわらず、`record_shot`させるとこうなります。

```suggestion
      frames.each do |frame|
        frame.add_bonus(shot_score.to_i)
        frame.record_shot(shot_score.to_i)
      end
```

この場合、`shot_score`を自分に追加するかどうかは、Frameオブジェクト自身が判断します。
（お腹が空いてたら食べる、満腹なら食べない、みたいな感じ）

そこからさらに考えを進めると・・・（ここから先はsekitoさんの宿題にしますね）

Reply:
宿題の回答としてあっているかわかりませんが、`record_shot`メソッド内だけでインスタンス変数の変更が完結するように修正してみました

Reply:
> 宿題の回答としてあっているかわかりませんが

合ってます！ 🙆‍♂️